types/react-native/index.d.ts

```diff
@@ -8939,24 +8939,39 @@ export namespace Animated {
 
     export type LegacyRef<C> = { getNode(): C };
 
-    export interface WithAnimatedValue<T>
-        extends ThisType<
-            T extends object
-                ? { [K in keyof T]?: WithAnimatedValue<T[K]> }
-                : T extends (infer P)[]
-                ? WithAnimatedValue<P>[]
-                : T | Value | AnimatedInterpolation
-        > {}
-
-    export type AnimatedProps<T> = { [key in keyof T]: WithAnimatedValue<T[key]> } &
-        (T extends {
-            ref?: React.Ref<infer R>;
-        }
-            ? { ref?: React.Ref<R | { getNode(): R }> }
-            : {});
+    type Nullable = undefined | null;
+    type Primitive = string | number | boolean | symbol;
+    type Builtin = Function | Date | Error | RegExp;
+
+    interface WithAnimatedArray<P> extends Array<WithAnimatedValue<P>> {}
+    type WithAnimatedObject<T> = {
+        [K in keyof T]: WithAnimatedValue<T[K]>;
+    };
+
+    export type WithAnimatedValue<T> = T extends Builtin | Nullable
+        ? T
+        : T extends Primitive // add `Value` and `AnimatedInterpolation` but also preserve original T
+        ? T | Value | AnimatedInterpolation
+        : T extends Array<infer P>
+        ? WithAnimatedArray<P>
+        : T extends {}
+        ? WithAnimatedObject<T> // in case it's something we don't yet know about (for .e.g bigint)
+        : T;
+
+    type NonAnimatedProps = 'key' | 'ref';
```

Comment:
I'm not really sure we should augment all props with `Value and AnimatedInterpolation`. 
I tried to find any kind of useful information in docs or react-native source code where `Animated.Value` and `AnimatedInterpolation` can be passed as a prop (except `style` prop) to narrow augmentation but I just don't see any kind of list of supported by animated values props.